src/dune_lang/package_name.ml

```diff
@@ -10,10 +10,52 @@ include (
     let description = "package name"
     let description_of_valid_string = None
     let hint_valid = None
-
-    let of_string_opt s =
-      (* DUNE3 verify no dots or spaces *)
-      if s = "" then None else Some s
-    ;;
+    let of_string_opt s = if s = "" then None else Some s
   end) :
     Dune_util.Stringlike with type t := t)
+
+module Strict = struct
```

Comment:
I'm not sure this is a good name. It is a stricter version than what dune-packages can be but the usecase would be opam compatibility, so maybe `Opam_compatible`?